src/machine/mod.rs

```diff
@@ -417,6 +417,7 @@ pub struct DefaultMachine<Inner> {
     debugger: Option<Box<dyn Debugger<Inner>>>,
     syscalls: Vec<Box<dyn Syscalls<Inner>>>,
     exit_code: i8,
+    peak_memory_usage: usize,
```

Comment:
It is recommended to rename it to `memory_size`, `peak_memory_usage` is a concept in `spawn`.